Week1task6/src/ArraySearch.java

```diff
@@ -0,0 +1,27 @@
+import java.util.Arrays;
+public class ArraySearch {
+
+	public static void main(String[] args) {
+		//initialising an array
+		int[] number = {38, 15, 42, 78, 49, 11, 61, 19, 90, 3};
+		System.out.println("Unsorted Array: " + Arrays.toString(number));
+		
+		// sorting array
+		Arrays.sort(number);
+		System.out.println("Sorted Array: " + Arrays.toString(number));
+		
+		// searching an existing value in the array
+		int searchValue = 49;
+		int foundValue = Arrays.binarySearch(number, searchValue);
+		System.out.println("Searching existing value: " + searchValue);
+		System.out.println("Found value: " + number[foundValue]);
+		
+		// searching an unexisting value in the array
+		int searchValue2 = 2;
+		int foundValue2 = Arrays.binarySearch(number, searchValue2);
```

Comment:
Because the value 2 does not exist in the array, the program will have an exception: ArrayIndexOutOfBounds. This is because foundValue2 = -1 and the array does not have this index. Try adding an if before calling binarySearch and test if the returned value is different than -1. Only then, if foundValue is different than -1 you can print number[foundValue]. In general, it is not good to have exceptions. Exceptions make the program to stop running and this is not good for the users.